### ISSUE TITLE
Partition processing stops and never comes back

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe#5127]
**Describe the bug**

Every week i create a benchmark on the ChaosDay and let it run until the next Chaos day. Today I investigated why the processing went down. I will collect here my observations and use this more as a summary, since until now it is not clear what is the issue.

**Partition 1 last 7 days**
![partition-1-general-7-days](https://user-images.githubusercontent.com/2758593/89543087-cb433380-d800-11ea-9d03-27d60a6cc49d.png)
**Partition 2 last 7 days**
![partition-2-general-7-days](https://user-images.githubusercontent.com/2758593/89543091-cb433380-d800-11ea-80ab-922c483cd039.png)
**Partition 3 last 7 days**
![partition-3-general-7-days](https://user-images.githubusercontent.com/2758593/89543093-cbdbca00-d800-11ea-9e86-d14b165e02eb.png)

We can see that the processing seem to stopped for the partition two and never comes back.This seem to happen on the 31-07-2020 ~ 2 pm.
With the resource panel we can also see that there happen a node preemption, since all pod seem to be rescheduled.
![resources](https://user-images.githubusercontent.com/2758593/89543095-cbdbca00-d800-11ea-962e-aee3fa95a826.png)

In the log we can see that all brokers are getting closed.

Interesting is that we see the day after also an increasing of disk usage, which never gets reduced.
![disk-usage](https://user-images.githubusercontent.com/2758593/89543083-c9797000-d800-11ea-82f3-a1b505811171.png)

Currently I don't understand what was the issue and why it never comes back.

**To Reproduce**

I will setup a new cluster to try to reproduce this.

**Expected behavior**
Running without issues.

**Log/Stacktrace**

[log.txt](https://github.com/zeebe-io/zeebe/files/5035475/log.txt)
[query.txt](https://github.com/zeebe-io/zeebe/files/5035477/query.txt)

https://console.cloud.google.com/logs/viewer?interval=JUMP_TO_TIME&authuser=1&organizationId=669107107215&project=zeebe-io&minLogLevel=0&expandAll=false&timestamp=2020-07-31T13:50:46.000000000Z&customFacets=jsonPayload.context.actor-name&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22zeebe-chaos%22&scrollTimestamp=2020-07-31T13:50:45.912808198Z&dateRangeStart=2020-07-31T12:50:46.000Z&dateRangeEnd=2020-07-31T14:50:46.000Z

**Environment:**
 - OS: k8s
 - Zeebe Version: SNAPSHOT
 - Configuration: default benchmark

**Comment by Zelldon** [comment]
I run a second benchmark and was not able to reproduce it.

![reprocess](https://user-images.githubusercontent.com/2758593/89757358-4cdae000-dae5-11ea-86a4-2a1cd58e5c4e.png)

**Comment by Zelldon** [comment]
### Investigation Benchmark Zeebe-Chaos

I will first summarize what we know and what we see based on the metrics. Later I will all things I can find out during the investigation.

In the general overview we see a lower throughput then usual. 
![otherpartitions](https://user-images.githubusercontent.com/2758593/89760795-62ec9e80-daed-11ea-9ad6-53ba6f010854.png)

If we take a specific look at the partition 2 then we can see that this partition died quite early.
![partition2-general](https://user-images.githubusercontent.com/2758593/89760802-64b66200-daed-11ea-935e-3d723136a509.png)

Interesting is that we see a lot of leader changes during the lifetime of the benchmark.

![p2-atomix](https://user-images.githubusercontent.com/2758593/89760797-641dcb80-daed-11ea-964a-b3c651729ae1.png)

If we take a look at the throughput panel we can see that it seems that at least reprocessing seem to work.
![p2-reprocessing](https://user-images.githubusercontent.com/2758593/89760798-641dcb80-daed-11ea-8e34-472399b72b6f.png)

The start up metrics shows that the starting takes ~ 45 min.

![p2-startup](https://user-images.githubusercontent.com/2758593/89760799-64b66200-daed-11ea-9a3d-164d2ff7e966.png)

If we request the topology on the gateway we can see that the leader is correctly distributed via gossip. Broker 1 is currently leader for partition 2.

```
first record does not look like a TLS handshake"
root@zeebe-chaos-zeebe-gateway-75d97f6478-snwgs:/usr/local/zeebe# zbctl status --insecure
Cluster size: 3
Partitions count: 3
Replication factor: 3
Gateway version: 0.25.0-SNAPSHOT
Brokers:
  Broker 0 - zeebe-chaos-zeebe-0.zeebe-chaos-zeebe.zeebe-chaos.svc.cluster.local:26501
    Version: 0.25.0-SNAPSHOT
    Partition 1 : Follower
    Partition 2 : Follower
    Partition 3 : Leader
  Broker 1 - zeebe-chaos-zeebe-1.zeebe-chaos-zeebe.zeebe-chaos.svc.cluster.local:26501
    Version: 0.25.0-SNAPSHOT
    Partition 1 : Leader
    Partition 2 : Leader
    Partition 3 : Follower
  Broker 2 - zeebe-chaos-zeebe-2.zeebe-chaos-zeebe.zeebe-chaos.svc.cluster.local:26501
    Version: 0.25.0-SNAPSHOT
    Partition 1 : Follower
    Partition 2 : Follower
    Partition 3 : Follower
root@zeebe-chaos-zeebe-gateway-75d97f6478-snwgs:/usr/local/zeebe# [zell ~/ ns:zeebe-chaos]$ 
```

This also means that the clients should be able to send commands to the partition. Then something seems to be broken in the processing. We need to take a deeper look at the logs.

**Comment by Zelldon** [comment]
### Investigate Logs

#### Starter

The starter sees a lot of failed requests, which makes sense when partition two is dead.

Unfortunately it is not clear on the client, what the error causes and what the problem is.

```
2020-08-10T08:49:29.907041471Z java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: INTERNAL: Unexpected error occurred during the request processing
	at java.util.concurrent.CompletableFuture.reportGet(Unknown Source) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(Unknown Source) ~[?:?]
	at io.zeebe.ResponseChecker.run(ResponseChecker.java:41) [app.jar:0.23.3]
Caused by: io.grpc.StatusRuntimeException: INTERNAL: Unexpected error occurred during the request processing
	at io.grpc.Status.asRuntimeException(Status.java:533) ~[app.jar:0.23.3]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:449) ~[app.jar:0.23.3]
	at io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39) ~[app.jar:0.23.3]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23) ~[app.jar:0.23.3]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) ~[app.jar:0.23.3]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) ~[app.jar:0.23.3]
	at io.grpc.internal.ClientCallImpl.access$500(ClientCallImpl.java:66) ~[app.jar:0.23.3]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:689) ~[app.jar:0.23.3]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$900(ClientCallImpl.java:577) ~[app.jar:0.23.3]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:751) ~[app.jar:0.23.3]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:740) ~[app.jar:0.23.3]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[app.jar:0.23.3]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123) ~[app.jar:0.23.3]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	at java.lang.Thread.run(Unknown Source) ~[?:?]
 I 
  undefined
2020-08-10T08:49:29.907119497Z 08:49:29.906 [Thread-2] WARN  io.zeebe.ResponseChecker - Request failed
 I 
  undefined
```

**Comment by Zelldon** [comment]
### Investigating Broker Logs

On opening the Broker logs I directly saw some weird messages.

#### Weird class loading message

```
I 2020-08-10T09:00:43.034371344Z 1177:47  WARN: [kryo] Unable to load class  e with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.035851662Z 1177:47  WARN: [kryo] Unable to load class  f with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.036390170Z 1177:47  WARN: [kryo] Unable to load class  g with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.037425588Z 1177:47  WARN: [kryo] Unable to load class  h with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.038462281Z 1177:47  WARN: [kryo] Unable to load class  i with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.039447158Z 1177:47  WARN: [kryo] Unable to load class  j with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.040062484Z 1177:47  WARN: [kryo] Unable to load class  k with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.040568978Z 1177:47  WARN: [kryo] Unable to load class HGl with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.041273564Z 1177:47  WARN: [kryo] Unable to load class px7 with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.042711536Z 1177:47  WARN: [kryo] Unable to load class 7hb7s with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.043388448Z 1177:47  WARN: [kryo] Unable to load class 7hb7t with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.043618996Z 1177:47  WARN: [kryo] Unable to load class 7hb7u with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.044744082Z 1177:47  WARN: [kryo] Unable to load class 7hb7v with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.045427433Z 1177:47  WARN: [kryo] Unable to load class 7hb7w with kryo's ClassLoader. Retrying with current..
 
I 2020-08-10T09:00:43.046031821Z 1177:47  WARN: [kryo] Unable to load class 68 with kryo's ClassLoader. Retrying with current..
 
```
![output](https://user-images.githubusercontent.com/2758593/89766859-1c04a600-daf9-11ea-901b-67447e5f47d8.png)


If we check the labels of the messages then we see that they happen on the Broker 1, which is the current leader of the partition 2. This might be the cause of the current issue.

```

2020-08-10 11:00:43.040 CEST
1177:47 WARN: [kryo] Unable to load class ���HG�������������l with kryo's ClassLoader. Retrying with current..
 
Expand all | Collapse all{
 insertId: "j6dql08m5o4cgb2y5"  
 labels: {
  k8s-pod/app_kubernetes_io/component: "broker"   
  k8s-pod/app_kubernetes_io/instance: "zeebe-chaos"   
  k8s-pod/app_kubernetes_io/managed-by: "Helm"   
  k8s-pod/app_kubernetes_io/name: "zeebe-cluster"   
  k8s-pod/app_kubernetes_io/version: "0.23.4"   
  k8s-pod/controller-revision-hash: "zeebe-chaos-zeebe-674fc5df4b"   
  k8s-pod/helm_sh/chart: "zeebe-cluster-0.0.128"   
  k8s-pod/statefulset_kubernetes_io/pod-name: "zeebe-chaos-zeebe-1"   
 }
 logName: "projects/zeebe-io/logs/stdout"  
 receiveTimestamp: "2020-08-10T09:00:45.112627609Z"  
 resource: {…}  
 severity: "INFO"  
 textPayload: "1177:47  WARN: [kryo] Unable to load class HGl with kryo's ClassLoader. Retrying with current..
"  
 timestamp: "2020-08-10T09:00:43.040568978Z"  
}
```

**Comment by Zelldon** [comment]
### Interesting findings

It seems that the helm chart uses a different version as label as the current deployed version.


```
[zell ~/ ns:zeebe-chaos]$ k describe pod zeebe-chaos-zeebe-1 
Name:           zeebe-chaos-zeebe-1
Namespace:      zeebe-chaos
Priority:       0
Node:           gke-zeebe-cluster-default-pool-1f44de48-lzr8/10.132.0.66
Start Time:     Fri, 07 Aug 2020 12:59:14 +0200
Labels:         app.kubernetes.io/component=broker
                app.kubernetes.io/instance=zeebe-chaos
                app.kubernetes.io/managed-by=Helm
                app.kubernetes.io/name=zeebe-cluster
                app.kubernetes.io/version=0.23.4
                controller-revision-hash=zeebe-chaos-zeebe-674fc5df4b
                helm.sh/chart=zeebe-cluster-0.0.128
                statefulset.kubernetes.io/pod-name=zeebe-chaos-zeebe-1
Annotations:    <none>
Status:         Running
IP:             10.0.17.10
IPs:            <none>
Controlled By:  StatefulSet/zeebe-chaos-zeebe
Containers:
  zeebe-cluster:
    Container ID:   docker://67c27fe5382a704053b73ba439c5fe6fd7f105df6724628804c203eba2c0da9c
    Image:          camunda/zeebe:SNAPSHOT
    Image ID:       docker-pullable://camunda/zeebe@sha256:6dcbec81922b42653bfbe75cd5b2199328487927c5baaf006d3f12c10f957e16
    Ports:          9600/TCP, 26501/TCP, 26502/TCP
    Host Ports:     0/TCP, 0/TCP, 0/TCP
    State:          Running
      Started:      Sun, 09 Aug 2020 15:22:51 +0200
    Last State:     Terminated
      Reason:       Error
      Exit Code:    3
      Started:      Sun, 09 Aug 2020 13:03:30 +0200
      Finished:     Sun, 09 Aug 2020 15:22:48 +0200
    Ready:          True
    Restart Count:  1

```

**Comment by Zelldon** [comment]
### Disk usage

I reported in the beginning that the cluster has a large disk usage, it seems that this is related to partition 2. 
Partition two is the only partition which has a lot of segments.

![number-of-log-segments](https://user-images.githubusercontent.com/2758593/89776093-77d72b00-db09-11ea-9b95-ab93c8654094.png)

**Comment by Zelldon** [comment]
### Discussion

Did a short discussion with @MiguelPires about the logs we have seen here https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671242640 . 

The problem with our benchmark is that we use `camunda:SNAPSHOT` and `pullPolicy: Always` which means it will deploy a newer snapshot if the broker was restarted or node was  preempted. This means that we have the last changes in our running benchmarks which have been merged later. The benchmark was setup ~10 days again the PR https://github.com/zeebe-io/zeebe/pulls?q=is%3Apr+is%3Aclosed+author%3AMiguelPires was merged 7 days ago.

If we use some stackdriver magic then we can see that this log happens after the PR was merged.

![log-kryo](https://user-images.githubusercontent.com/2758593/89780315-7f9acd80-db11-11ea-8df5-2a8ee278951c.png)

He is able to reproduce it with a new Benchmark and will take a closer look. He think this is not a real issue and means it is expected if we read/receive old messages. If we take a look at https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671290156, then we see that we have a lot of old data in our log.

**Comment by Zelldon** [comment]
### Investigation OOM

In https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671242640 I found an indication of an OOM Error. After taking a look at the nodes, I saw that all of them had already some heap dump files.

After checking the heap dump we can see that this is due to many journal segments. 
This we have also seen here https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671290156.

![heap-bytearr](https://user-images.githubusercontent.com/2758593/89782836-43b63700-db16-11ea-901a-aa29e8cda78f.png)
![heap-journal](https://user-images.githubusercontent.com/2758593/89782838-44e76400-db16-11ea-8e73-8c4846a26cb5.png)

Each segment has at least one reader and writer, which means 16 MB buffer. Multiplying with 141 segments we get ~ 2 gig of mem usage. This seems to cause the OOM. The heap dump as also a size of ~ 2 gig. Weird that it is not larger :thinking:

**Comment by Zelldon** [comment]
Checking last appering errors via Stackdriver I found [this](https://console.cloud.google.com/logs/viewer?resource=k8s_container%2Fcluster_name%2Fzeebe-cluster%2Fnamespace_name%2Fzeebe-chaos%2Fcontainer_name%2Fzeebe-cluster&logName=projects%2Fzeebe-io%2Flogs%2Fstdout&expandAll=false&timestamp=2020-08-11T05:10:34.669000000Z&project=zeebe-io&authuser=1&minLogLevel=0&customFacets=&limitCustomFacetWidth=true&advancedFilter=error_group(%22CLXBrJHsz7fFfg%22)&interval=JUMP_TO_TIME&scrollTimestamp=2020-08-11T05:10:04.669271000Z):

```
E 2020-07-20T01:47:17.613500Z Actor Broker-2-StreamProcessor-1 failed in phase STARTED. 
E 2020-07-25T15:51:14.897973Z Actor Broker-2-StreamProcessor-1 failed in phase STARTED. 
E 2020-07-25T15:55:40.150140Z Actor Broker-2-StreamProcessor-3 failed in phase STARTED. 
E 2020-07-25T18:07:12.466330Z Actor Broker-1-StreamProcessor-1 failed in phase STARTED. 
E 2020-07-25T19:05:13.824363Z Actor Broker-1-StreamProcessor-1 failed in phase STARTED. 
E 2020-07-26T00:58:09.414511Z Actor Broker-1-StreamProcessor-1 failed in phase STARTED. 
E 2020-07-26T10:24:10.074232Z Actor Broker-0-StreamProcessor-1 failed in phase STARTED. 
E 2020-08-01T23:15:12.378426Z Actor Broker-2-StreamProcessor-2 failed in phase STARTED. 
E 2020-08-02T23:31:45.105097Z Actor Broker-2-StreamProcessor-2 failed in phase STARTED. 
E 2020-08-04T00:28:46.666387Z Actor Broker-0-StreamProcessor-2 failed in phase STARTED. 
E 2020-08-06T06:04:03.299869Z Actor Broker-1-StreamProcessor-2 failed in phase STARTED. 
E 2020-08-06T20:39:56.653116Z Actor Broker-2-StreamProcessor-2 failed in phase STARTED. 
E 2020-08-07T02:47:06.717342Z Actor Broker-1-StreamProcessor-1 failed in phase STARTED. 
E 2020-08-08T19:51:57.202311Z Actor Broker-2-StreamProcessor-2 failed in phase STARTED. 
2020-08-10T09:31:01.538472Z Actor Broker-1-StreamProcessor-2 failed in phase STARTED. E 
2020-08-11T05:10:04.669271Z Actor Broker-0-StreamProcessor-2 failed in phase STARTED. E 
```

Details:
```
{
 insertId: "4ndl93dz0w03mids6"  
 jsonPayload: {
  @type: "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent"   
  context: {…}   
  exception: "java.lang.NullPointerException: null
	at io.zeebe.engine.processing.job.JobTimeoutTrigger.onResumed(JobTimeoutTrigger.java:63) ~[zeebe-workflow-engine-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at java.util.ArrayList.forEach(Unknown Source) ~[?:?]
	at io.zeebe.engine.processing.streamprocessor.StreamProcessor.lambda$resumeProcessing$4(StreamProcessor.java:354) ~[zeebe-workflow-engine-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorControl.lambda$call$0(ActorControl.java:151) ~[zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:62) ~[zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:118) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:204) [zeebe-util-0.25.0-SNAPSHOT.jar:0.25.0-SNAPSHOT]
"   
  logger: "io.zeebe.logstreams"   
  message: "Actor Broker-2-StreamProcessor-2 failed in phase STARTED."   
  serviceContext: {…}   
  thread: "Broker-2-zb-actors-0"   
 }
 labels: {…}  
 logName: "projects/zeebe-io/logs/stdout"  
 receiveTimestamp: "2020-08-02T23:32:15.764714308Z"  
 resource: {…}  
 severity: "ERROR"  
 sourceLocation: {…}  
 timestamp: "2020-08-02T23:31:45.105097Z"  
}
```

**Comment by Zelldon** [comment]
### Continue Log Investigation

If we take again a closer look at the metrics we can see when new brokers become leader for partition 2.
We can see that actually only exporting seems to happen, but no real processing.

![p2-general](https://user-images.githubusercontent.com/2758593/89870972-f50ca980-dbb6-11ea-9096-b5751b8235ee.png)

If we check the log for example at [2020-08-08 18:00](https://console.cloud.google.com/logs/viewer?interval=JUMP_TO_TIME&authuser=1&project=zeebe-io&minLogLevel=0&expandAll=false&timestamp=2020-08-08T16:00:00.000000000Z&customFacets=labels.%22k8s-pod%2Fstatefulset_kubernetes_io%2Fpod-name%22,jsonPayload.context.actor-name&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22zeebe-chaos%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fcomponent%3D%22broker%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Finstance%3D%22zeebe-chaos%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fmanaged-by%3D%22Helm%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fname%3D%22zeebe-cluster%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fversion%3D%220.23.4%22%0Alabels.k8s-pod%2Fhelm_sh%2Fchart%3D%22zeebe-cluster-0.0.128%22%0AjsonPayload.logger!%3D%22io.zeebe.broker.exporter.elasticsearch%22%0AjsonPayload.logger!%3D%22io.atomix.raft.roles.LeaderAppender%22&scrollTimestamp=2020-08-08T19:51:57.079807000Z&dateRangeUnbound=both) then we can see the following:

```
W 2020-08-08T19:50:21.256529Z zeebe-chaos-zeebe-2 io.zeebe.broker.system.monitoring.DiskSpaceUsageMonitor Out of disk space. Current available 26794422272 bytes. Minimum needed 26841658982 bytes.  zeebe-chaos-zeebe-2 io.zeebe.broker.system.monitoring.DiskSpaceUsageMonitor
W 2020-08-08T19:50:21.267990Z zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-2 Disk space usage is above threshold. Pausing stream processor.  zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-2
W 2020-08-08T19:50:21.268822Z zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-3 Disk space usage is above threshold. Pausing stream processor.  zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-3
E 2020-08-08T19:50:42.031027Z zeebe-chaos-zeebe-2 Broker-2-HealthCheckService Partition-2 failed, marking it as unhealthy  zeebe-chaos-zeebe-2 Broker-2-HealthCheckService
E 2020-08-08T19:50:44.766158Z zeebe-chaos-zeebe-2 Broker-2-HealthCheckService Partition-3 failed, marking it as unhealthy  zeebe-chaos-zeebe-2 Broker-2-HealthCheckService
I 2020-08-08T19:51:56.423357Z zeebe-chaos-zeebe-2 Broker-2-SnapshotDirector-3 Finished taking snapshot, need to wait until last written event position 763960499 is committed, current commit position is 763960499. After that snapshot can be marked as valid.  zeebe-chaos-zeebe-2 Broker-2-SnapshotDirector-3
I 2020-08-08T19:51:56.705888Z zeebe-chaos-zeebe-2 Broker-2-SnapshotDirector-3 Current commit position 763960499 is greater then 763960499, snapshot DbSnapshotMetadata{index=602289899, term=222, timestamp=2020-08-08 07:51:55,549} is valid and has been persisted.  zeebe-chaos-zeebe-2 Broker-2-SnapshotDirector-3
I 2020-08-08T19:51:57.076949Z zeebe-chaos-zeebe-2 io.zeebe.broker.system.monitoring.DiskSpaceUsageMonitor Disk space available again. Current available 33525628928 bytes  zeebe-chaos-zeebe-2 io.zeebe.broker.system.monitoring.DiskSpaceUsageMonitor
I 2020-08-08T19:51:57.079807Z zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-3 Disk space usage is below threshold. Resuming stream processor.  zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-3
I 2020-08-08T19:51:57.079858Z zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-2 Disk space usage is below threshold. Resuming stream processor.  zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-2
E 2020-08-08T19:51:57.202311Z zeebe-chaos-zeebe-2 Broker-2-StreamProcessor-2 Actor Broker-2-StreamProcessor-2 failed in phase STARTED.  zeebe-chaos-zeebe-2 Broker-2-StreamProcessor-2
E 2020-08-08T19:51:57.205164Z zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-2 Broker-2-StreamProcessor-2 failed, marking it as unhealthy  zeebe-chaos-zeebe-2 Broker-2-ZeebePartition-2
```

This problem we have seen already here https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671758191
It seems that the broker becomes leader, registered the Out of Disk Space listener and needs to pause, because it is out of disk. After another partition takes a snapshot the processor should be resumed but this failes with #5149 .

**Comment by Zelldon** [comment]
### Continue Log Investigation 

We found out that the partition never comes back because of a problem with the onResume in the JobTimeoutTrigger, see https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671795945

I wanted to find out why the disk went full in the first place and why the partition failed at the begin
[I checked the log and can clearly see that at 31-07-2020 ~2 pm all brokers are closed](https://console.cloud.google.com/logs/viewer?interval=JUMP_TO_TIME&authuser=1&project=zeebe-io&minLogLevel=0&expandAll=false&timestamp=2020-07-31T09:57:11.000000000Z&customFacets=labels.%22k8s-pod%2Fstatefulset_kubernetes_io%2Fpod-name%22,jsonPayload.context.actor-name&limitCustomFacetWidth=true&scrollTimestamp=2020-07-31T12:28:05.029019000Z&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22zeebe-chaos%22%0Alabels.%22k8s-pod%2Fapp%22!%3D%22worker%22%0AjsonPayload.context.loggerName!%3D%22io.zeebe.gateway%22%0ANOT%20jsonPayload.message:%22Waiting%20for%20more%20snapshot%20chunks%20of%22&dateRangeStart=2020-07-30T09:57:11.000Z&pinnedLogId=pugh0wm7p95n7zawh&pinnedLogTimestamp=2020-07-31T12:27:58.961855Z). We have mentioned that already here https://github.com/zeebe-io/zeebe/issues/5127#issue-674358195

After checking the [audit logs ](https://console.cloud.google.com/logs/viewer?interval=JUMP_TO_TIME&authuser=1&project=zeebe-io&minLogLevel=0&expandAll=false&timestamp=2020-07-31T12:43:36.000000000Z&customFacets=&limitCustomFacetWidth=true&advancedFilter=logName%3D%22projects%2Fzeebe-io%2Flogs%2Fcloudaudit.googleapis.com%252Factivity%22%0Aresource.type%3D%22k8s_cluster%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.location%3D%22europe-west1-b%22%0AprotoPayload.request.metadata.name%3D%22zeebe-chaos-zeebe%22%0AprotoPayload.methodName:%22statefulsets.%22&dateRangeEnd=2020-07-31T13:43:36.000Z&scrollTimestamp=2020-07-31T12:29:24.539192000Z&dateRangeUnbound=backwardInTime)I found out that I recreated the cluster on 14:29 at this day. The exact moment when the brokers are closed.

If we take a look at the GKE resource panel's, then we see that immediately after the setup the disk usage seem to increase. This is also we saw before in our metrics. 
![gke-res](https://user-images.githubusercontent.com/2758593/89884302-a3b9e580-dbc9-11ea-83eb-049ffe897623.png)

**Comment by Zelldon** [comment]
### Summary

As usual several things coming together to paint this nice picture.

We have setup the benchmark on 31/07/2020 14:29, since the beginning the snapshotting failed, because the exporter failed to export 'cause of a broken pipe, see this https://github.com/zeebe-io/zeebe/issues/5127#issuecomment-671850388. There is already an open issue for that #4951. The snapshot fail message was unfortunately not clear enough to directly find this issue, see https://github.com/zeebe-io/zeebe/issues/5142. In the beginning this happen for all partitions, but later only for partition two.

Due to not exporting and not taking snapshot the disk usage grows very fast until we reach our disk usage threshold, where we stop accepting requests. Since this happens only for one partition the other partitions were still able to take snapshots and compact. Compacting freed the disk and we went from time to time under the disk usage threshold. This caused a different bug in resuming the StreamProcessor https://github.com/zeebe-io/zeebe/issues/5149, which means we failed to resume and were not able to continue with this partition.

Since 03/08/2020 we also have flood the log with new warn logs, which makes the debugging also a bit harder. https://github.com/zeebe-io/zeebe/issues/5146


Since the investigation is done for this benchmark and we created follow up issues for each of these problems I will close this issue. I would like to fix #5149 as soon as possible and also #5142 to make it more easier to understand why snapshotting actually doesn't work.

\cc @npepinpe
